### PR TITLE
Hr open preselections

### PR DESCRIPTION
**Pull request by alex-nayya**
> **_SHIPPED NOTE:_**  Add Preselections payload schema and endpoint to OpenAPI docs.

# Context
hr_open api docs didn't have the preselection payload schema defined. https://app.shortcut.com/nayya/story/26303/generate-open-api-hr-open-preselection-schema

# What this PR does
- Add the schema for preselections and all associated models.
- Add a POST endpoint 
- Descriptions for schema, endpoint, and models

# Test plan
- ensure there are no yaml errors
- review the title and description for each new model 

--- 

## As the author of this PR, I have (if applicable):

- [x] Tested my proposed changes and verified that there were no server errors or UI bugs
- [x] Assigned one or more [reviewers](https://www.notion.so/Code-Reviews-0c16b837cc4b418c904ed39691c193fe#2e862cf919f14c95937e4d373f218ad6) 
- [ ] If migrations are added, they are reversible. Annotations have been updated and seeds.rb has been updated, if relevant.
- [ ] Included translations for any text added
- [ ] Added any new environmental variables to AWS (github action), `locals.tf`, and 1Password as appropriate
- [ ] Added documentation in [notion](https://www.notion.so/Product-and-Engineering-e965393fa4154611b4b008d9adb08e86) describing how the feature works

### CODE REVIEW

File: models/hropen/base/IdentifierType.yaml [hunk 1]
```diff
@@ -2,13 +2,12 @@ title: IdentifierType
...
-  - value
 examples:
   - value: string
 x-tags:
   - hropen/base
+properties:
+  value:
+    anyOf: []
+required:
+  - value
```
Comment (licarpen) [inline models/hropen/base/IdentifierType.yaml:13]: is this an intentional change?

File: reference/PreselectionParams.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+properties:
+  source:
+    type: string
+  payload:
+    type: object
+    required:
+      - data
+    properties:
+      data:
+        $ref: ./PreselectionData.yaml
```
Comment (licarpen) [inline reference/PreselectionParams.yaml:14]: @blevine2317 this is the section we need Choose's input on - is this the structure of the payload that is sent back to ADP with the recommendation/preselection payload at the end of the Choose survey?

File: models/hropen/base/IdentifierType.yaml (hunk 1 above)
Comment (alex-nayya) [reply models/hropen/base/IdentifierType.yaml:13]: Nope. Will be reverted back to original on next commit.

File: reference/decision-support-api-hropen.yaml [hunk 3]
```diff
@@ -292,6 +292,23 @@ paths:
...
+          description: OK
+      description: |-
+        ENDPOINT NOT EXPOSED - payload represented here is posted via iFrame - contains all data pertinent to an employee selections,
+
+        Inlucdes enrollment hash with document id, person, coverages, and HSA information.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: ./IndividualEnrollmentParams.yaml
```
[the remark this answers is not in the document]
Comment (alex-nayya) [reply reference/decision-support-api-hropen.yaml:310]: yes.

File: reference/PreselectionParams.yaml (hunk 2 above)
Comment (blevine2317) [reply reference/PreselectionParams.yaml:14]: Just reviewed the top-level quickly, I think its missing the most recent changes from Precious. They're documented at the bottom of this Notion doc: https://www.notion.so/Feature-Post-user-recommendations-earlier-in-ADP-WFN-journey-d6ace3f46ac44eb2a79f6a89db20b663 @alex-nayya Let me know if you'd like to sync to get those updates in there.

I'm gonna review the rest of the attributes more closely in a bit, will update here.

Comment (alex-nayya) [reply reference/PreselectionParams.yaml]: @blevine2317 I made a new push that incorporated the json schema from the notion doc you linked me to. ready for review

**Review by blevine2317 — APPROVED** [review]
This is looking good to me! Did a comparison between the code and the Stoplight docs and didn't see any discrepancies. Thanks for working on this!

**Pull request merged**